packages/camera/camera/lib/src/camera_controller.dart

```diff
@@ -529,6 +539,12 @@ class CameraController extends ValueNotifier<CameraValue> {
         'startVideoRecording was called when a recording is already started.',
       );
     }
+    if (value.captureMode == CaptureMode.photo) {
+      throw CameraException(
```

Comment:
Is there a reason that you can't record a video in photo mode, but you can take a picture in video mode? Was this to avoid a breaking change?

Reply:
A lot of the different platforms have different settings and optimisations than can be enabled for still photo capture. If we guarantee not being able to take a video when the capture mode is photo we can make better use of this.

Taking a picture in video mode is acceptable since it's a common function in native cameras and as far as I'm aware it's mostly just a still from the video feed (There could be further optimisations here).